sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\HttpFactory;
+
+class ExporterFactory
+{
+    private $name;
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+      * Selects the correct Exporter via the configuration string
+      * Currently only supports Jaeger and Zipkin exporters
+      *
+      * @param string $configurationString String containing unextracted information for Exporter creation
+      */
+    public function fromConnectionString(string $configurationString)
+    {
+        $strArr = explode('+', $configurationString);
+        $contribName = $strArr[0];
```

Comment:
This should be validated against an allowlist.  This is untrusted user input and needs to be validated.  Since we know the allowable values here an allowlist should be used to reject any non-allowed values.  My PHP is incredibly rusty, but something like this should work:

```php
private $_allowedExporters = array("Jaeger" => true, "Zipkin" => true);

private function _isAllowed(string $exporter)
{
  return array_key_exists($this->_allowedExporters, $exporter) && $this->_allowedExporters{$exporter};
}

...
$contribName = ucfirst(lower($strArr[0]));
if !$this->_isAllowed($contribName) {
  return null // or some way to indicate an error here
}
$dynamicExporter = ...
```

Reply:
Also look into the htmlentities() function which could serve as a way to sanitize input. Make sure it is relevant to your issue, usually it is used mostly to protect from front end malicious input.

Reply:
Thanks for these reviews. Will soon have an update to reflect this